.github/workflows/main.yml

```diff
@@ -0,0 +1,78 @@
+name: Test Build and Call With Fake Register
+
+on: [push]
+
+jobs:
+  ExpectedConnectionReset:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python 3.x
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y openssl
+
+      - name: Install Python Dependencies
+        run: |
+          python3 -m pip install --upgrade pip
+          pip3 install requests
+
+      - name: Create Dummy Certificates and Keys
+        run: |
+          mkdir -p dummy-certs
+          openssl req -new -newkey rsa:2048 -days 365 -nodes -x509 -keyout dummy-certs/dummy.key -out dummy-certs/dummy.crt -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=www.example.com"
+
+      - name: Build Docker Compose
+        run: |
+          docker-compose build
+
+      - name: Prepare Certificates
+        run: |
+          mkdir -p ./factory-registration-ref/data/certs
+          cp dummy-certs/dummy.crt ./factory-registration-ref/data/certs/factory_ca.pem
+          cp dummy-certs/dummy.crt ./factory-registration-ref/data/certs/local-ca.pem
+          cp dummy-certs/dummy.key ./factory-registration-ref/data/certs/local-ca.key
+
+      - name: Run Docker Compose
+        run: |
+          docker-compose up -d
+
+      - name: Execute Script with Expectation of ConnectionResetError
+        run: |
+          set +e
+            output=$(python3 fake-lmp-device-register --registration-url "http://localhost:80/sign" --factory fake-test-sample 2>&1)
+            exit_code=$?
+          set -e
+          if [ $exit_code -ne 0 ]; then
+            echo "Non-zero exit code: $exit_code"
+            if echo "$output" | grep -q 'ConnectionResetError'; then
+              echo "Expected connection reset failure occurred."
+            else
+              echo "Unexpected failure occurred."
+              exit 1
+            fi
+          fi
+          echo "Output:"
+          echo "$output"
+
+      - name: Check relative /var/sota directory and pkey.pem file
+        run: |
+          if [[ -d "var/sota" && -f "var/sota/pkey.pem" ]]; then
+            echo "Relative /var/sota directory exists and pkey.pem is present"
```

Comment:
It must be present because we are calling the fake register here.
Therefore, we will create the dir and the key prior call the host to register the server.
The connection rest from the server.

Reply:
I don't get why is the connection being reset. Why would that error cannot be an expected outcome?
There is something wrong with the test setup; looks like bad timing or something e.g. the server did not yet come up.

Reply:
Same: https://github.com/foundriesio/factory-registration-ref/pull/17#discussion_r1382355542

Anyway, I changed the mock certs and sorted out the issue to ensure that it fails when we call in the app.py:
```python
for x in (0.1, 0.2, 1, 0):
        r = requests.put(
            "https://api.foundries.io/ota/devices/", headers=headers, json=data
        )
```